app/forms/training_details_form.rb

```diff
@@ -16,10 +16,13 @@ class TrainingDetailsForm
                          },
                          if: -> { validate_trainee_id }
 
-  validate :commencement_date_valid, if: -> { validate_commencement_date }
+  validate :commencement_date_valid, if: -> { training_route_manager.requires_start_date? && validate_commencement_date }
```

Comment:
It looks good, I'm wondering should, `TrainingRouteManager` sit inside `Trainee` or even `ProgresService` since it looks a bit tied into the validation related items.

Reply:
I like that idea @defong. I've pushed another commit to demonstrate what it could look like if we use the `Trainee` mode.